exotations/solvers/exotica_ilqr_solver/src/ilqr_solver.cpp

```diff
@@ -42,32 +42,32 @@ void ILQRSolver::SpecifyProblem(PlanningProblemPtr pointer)
     MotionSolver::SpecifyProblem(pointer);
     prob_ = std::static_pointer_cast<DynamicTimeIndexedShootingProblem>(pointer);
     dynamics_solver_ = prob_->GetScene()->GetDynamicsSolver();
-    if (debug_) HIGHLIGHT_NAMED("ILQRSolver", "initialized");
 }
 
 void ILQRSolver::BackwardPass()
 {
-    constexpr double min_clamp_ = -1e10;
-    constexpr double max_clamp_ = 1e10;
+    // constexpr double min_clamp_ = -1e10;
+    // constexpr double max_clamp_ = 1e10;
     const int T = prob_->get_T();
     const double dt = dynamics_solver_->get_dt();
 
-    const Eigen::MatrixXd Qf = prob_->get_Qf(), R = dt * prob_->get_R();
-    const Eigen::MatrixXd X_star = prob_->get_X_star();
+    const Eigen::MatrixXd& Qf = prob_->get_Qf();
+    const Eigen::MatrixXd R = dt * prob_->get_R();
+    const Eigen::MatrixXd& X_star = prob_->get_X_star();
 
     // eq. 18
     vk_gains_[T - 1] = Qf * dynamics_solver_->StateDelta(prob_->get_X(T - 1), X_star.col(T - 1));
-    vk_gains_[T - 1] = vk_gains_[T - 1].unaryExpr([min_clamp_, max_clamp_](double x) -> double {
-        return std::min(std::max(x, min_clamp_), max_clamp_);
-    });
+    // vk_gains_[T - 1] = vk_gains_[T - 1].unaryExpr([min_clamp_, max_clamp_](double x) -> double {
```

Comment:
This clamping was from Emo's code here: https://homes.cs.washington.edu/~todorov/software/iLQG.zip

Seems sensible... why did you remove it?